### ISSUE TITLE
Q value is always zero for any new model I train

### ISSUE THREAD

**Issue by amritkochar** [BerkeleyAutomation/dex-net#43]
Hello team,

When I use the GQ-Image-Wise model, for each grasp the q value is generated very well and according to the grasp quality. But, if I retrain or fine-tune a new model with some modifications, the q value for any grasp is always zero.

Why? Please, help me in solving this.

**Comment by jeffmahler** [comment]
Hello, please provide more details. Have you looked at the raw q_values in the numpy array output by the data generation script? It's possible that they are quite small and you will need to change the training threshold.

**Comment by amritkochar** [comment]
Hello Jeff,

If I understood correctly, I went to the dataset generated by running generate_gqcnn_dataset.py, and in the tensors folder when I check the values named robust_ferrari_canny_00XXX.npz, most of the values are very very small, and in all the np arrays, the max value is around 0.003-0.005.

In the generate_gqcnn_dataset.yaml file, these are the parameter values:

> # Dataset gen params
> images_per_stable_pose: 50
> stable_pose_min_p: 0.0


Also, when I use the generated dataset to train, in the training.yaml file, these are the variable values:

> target_metric_name: robust_ferrari_canny
> metric_thresh: 0.002

Let me know what am I doing wrong.

Thanks!

**Comment by amritkochar** [comment]
Hello team,

Anything on this? I tried to change a few of the values but it still doesn't help solve the problem.

Thanks.

**Comment by jeffmahler** [comment]
@amrit-007 Based on the information you provided, I suspect that the grasp metrics are being computed as expected and you are using difficult objects. Usually, the max is about 0.005 for a dataset.

There are some issues when training with a small number of positive examples, which is why you may be seeing all zeros. Here are a few potential fixes:
1) Increase the batch size to 256 or greater
2) Reduce the "metric_thresh"

**Comment by AbdulghaniAltaweel** [comment]
I am trying to generate training data for GQ-CNN.  I got small values of robust_ferrari_canny_metric like 2e-6. To check where is the Mistake I tried the dexnet-code on the example and dexnet_2 databases and recompute the metric for some objects and got small values too. so Any suggestions ?
